### PR TITLE
Shotwell custom launcher

### LINKED ISSUE #193: shotwell does not build
shotwell does not build any more:
```
valac --target-glib=2.30.0 -g --enable-checking --fatal-warnings --ccode --enable-deprecated \
	--vapidir=../ --pkg=shotwell-plugin-dev-1.0 --pkg=gobject-2.0 --pkg=glib-2.0 --pkg=gdk-3.0 --pkg=gtk+-3.0 --pkg=gee-0.8 --pkg=libxml-2.0 --pkg=libsoup-2.4  \
	 \
	--vapidir=../common --pkg shotwell-plugin-common \
	--vapidir=../../vapi \
	shotwell-transitions.vala FadeEffect.vala SlideEffect.vala CrumbleEffect.vala BlindsEffect.vala CircleEffect.vala CirclesEffect.vala SquaresEffect.vala StripesEffect.vala ChessEffect.vala ClockEffect.vala
valac --target-glib=2.30.0 -g --enable-checking --fatal-warnings --ccode --enable-deprecated \
	--vapidir=../ --pkg=shotwell-plugin-dev-1.0 --pkg=gobject-2.0 --pkg=glib-2.0 --pkg=gdk-3.0 --pkg=gtk+-3.0 --pkg=gee-0.8 --pkg=gee-0.8 --pkg=libxml-2.0 --pkg=libsoup-2.4 --pkg=json-glib-1.0 --pkg=webkit2gtk-4.0  \
	 \
	--vapi=shotwell-plugin-common.vapi --header=shotwell-plugin-common.h \
	--vapidir=../../vapi \
	Resources.vala RESTSupport.vala
error: Package `shotwell-plugin-common' not found in specified Vala API directories or GObject-Introspection GIR directories
Compilation failed: 1 error(s), 0 warning(s)
../Makefile.plugin.mk:45: die Regel für Ziel „.stamp“ scheiterte
make[2]: *** [.stamp] Fehler 1
make[2]: Verzeichnis „/home/daniel/dev/snappy/snappy-playpen/shotwell/parts/shotwell/build/plugins/shotwell-transitions“ wird verlassen
Makefile:14: die Regel für Ziel „shotwell-transitions“ scheiterte
make[1]: *** [shotwell-transitions] Fehler 2
make[1]: *** Auf noch nicht beendete Prozesse wird gewartet …
cc -c -pthread -I/usr/include/gee-0.8 -I/usr/include/json-glib-1.0 -I/usr/include/webkitgtk-4.0 -I/usr/include/gtk-3.0 -I/usr/include/at-spi2-atk/2.0 -I/usr/include/at-spi-2.0 -I/usr/include/dbus-1.0 -I/usr/lib/x86_64-linux-gnu/dbus-1.0/include -I/usr/include/gtk-3.0 -I/usr/include/gio-unix-2.0/ -I/usr/include/mirclient -I/usr/include/mircommon -I/usr/include/mircookie -I/usr/include/cairo -I/usr/include/pango-1.0 -I/usr/include/harfbuzz -I/usr/include/pango-1.0 -I/usr/include/atk-1.0 -I/usr/include/cairo -I/usr/include/pixman-1 -I/usr/include/freetype2 -I/usr/include/libpng12 -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libpng12 -I/usr/include/libsoup-2.4 -I/usr/include/libxml2 -I/usr/include/webkitgtk-4.0 -I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include -nostdlib -export-dynamic -O2 -g -pipe  -fPIC -D_VERSION='"0.23.2"' -DGETTEXT_PACKAGE='"shotwell"' -I../.. -I ../common Resources.c
cc -c -pthread -I/usr/include/gee-0.8 -I/usr/include/json-glib-1.0 -I/usr/include/webkitgtk-4.0 -I/usr/include/gtk-3.0 -I/usr/include/at-spi2-atk/2.0 -I/usr/include/at-spi-2.0 -I/usr/include/dbus-1.0 -I/usr/lib/x86_64-linux-gnu/dbus-1.0/include -I/usr/include/gtk-3.0 -I/usr/include/gio-unix-2.0/ -I/usr/include/mirclient -I/usr/include/mircommon -I/usr/include/mircookie -I/usr/include/cairo -I/usr/include/pango-1.0 -I/usr/include/harfbuzz -I/usr/include/pango-1.0 -I/usr/include/atk-1.0 -I/usr/include/cairo -I/usr/include/pixman-1 -I/usr/include/freetype2 -I/usr/include/libpng12 -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libpng12 -I/usr/include/libsoup-2.4 -I/usr/include/libxml2 -I/usr/include/webkitgtk-4.0 -I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include -nostdlib -export-dynamic -O2 -g -pipe  -fPIC -D_VERSION='"0.23.2"' -DGETTEXT_PACKAGE='"shotwell"' -I../.. -I ../common RESTSupport.c
/home/daniel/dev/snappy/snappy-playpen/shotwell/parts/shotwell/build/plugins/common/RESTSupport.vala: In function ‘publishing_rest_support_argument_compare’:
/home/daniel/dev/snappy/snappy-playpen/shotwell/parts/shotwell/build/plugins/common/RESTSupport.vala:122:9: warning: assignment from incompatible pointer type [-Wincompatible-pointer-types]
         return strcmp(arg1.key, arg2.key);
         ^
/home/daniel/dev/snappy/snappy-playpen/shotwell/parts/shotwell/build/plugins/common/RESTSupport.vala: In function ‘publishing_rest_support_argument_sort’:
/home/daniel/dev/snappy/snappy-playpen/shotwell/parts/shotwell/build/plugins/common/RESTSupport.vala:131:9: warning: assignment from incompatible pointer type [-Wincompatible-pointer-types]
         return sorted_args.to_array();
         ^
/home/daniel/dev/snappy/snappy-playpen/shotwell/parts/shotwell/build/plugins/common/RESTSupport.vala: In function ‘publishing_rest_support_transaction_check_response’:
/home/daniel/dev/snappy/snappy-playpen/shotwell/parts/shotwell/build/plugins/common/RESTSupport.vala:218:10: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
         if (message.response_body.data == null || message.response_body.data.length == 0)
          ^
/home/daniel/dev/snappy/snappy-playpen/shotwell/parts/shotwell/build/plugins/common/RESTSupport.vala:218:11: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
         if (message.response_body.data == null || message.response_body.data.length == 0)
           ^
/home/daniel/dev/snappy/snappy-playpen/shotwell/parts/shotwell/build/plugins/common/RESTSupport.vala: In function ‘publishing_rest_support_transaction_get_response’:
/home/daniel/dev/snappy/snappy-playpen/shotwell/parts/shotwell/build/plugins/common/RESTSupport.vala:352:9: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
         return (string) message.response_body.data;
         ^
/usr/share/vala-0.30/vapi/glib-2.0.vapi: In function ‘string_chug’:
/usr/share/vala-0.30/vapi/glib-2.0.vapi:1109:13: warning: passing argument 1 of ‘g_strchug’ discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
   result._chug ();
             ^
In file included from /usr/include/glib-2.0/glib.h:79:0,
                 from RESTSupport.c:10:
/usr/include/glib-2.0/glib/gstrfuncs.h:175:8: note: expected ‘gchar * {aka char *}’ but argument is of type ‘const gchar * {aka const char *}’
 gchar*                g_strchug        (gchar        *string);
        ^
/usr/share/vala-0.30/vapi/glib-2.0.vapi: In function ‘string_chomp’:
/usr/share/vala-0.30/vapi/glib-2.0.vapi:1101:14: warning: passing argument 1 of ‘g_strchomp’ discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
   result._chomp ();
              ^
In file included from /usr/include/glib-2.0/glib.h:79:0,
                 from RESTSupport.c:10:
/usr/include/glib-2.0/glib/gstrfuncs.h:178:8: note: expected ‘gchar * {aka char *}’ but argument is of type ‘const gchar * {aka const char *}’
 gchar*                g_strchomp       (gchar        *string);
        ^
/home/daniel/dev/snappy/snappy-playpen/shotwell/parts/shotwell/build/plugins/common/RESTSupport.vala: In function ‘publishing_rest_support_google_publisher_web_authentication_pane_on_page_load’:
/home/daniel/dev/snappy/snappy-playpen/shotwell/parts/shotwell/build/plugins/common/RESTSupport.vala:761:2: warning: ‘gdk_cursor_new’ is deprecated [-Wdeprecated-declarations]
             pane_widget.get_window().set_cursor(new Gdk.Cursor(Gdk.CursorType.LEFT_PTR));
  ^
In file included from /usr/include/gtk-3.0/gdk/gdk.h:34:0,
                 from ../../plugins/shotwell-plugin-dev-1.0.h:15,
                 from RESTSupport.c:15:
/usr/include/gtk-3.0/gdk/gdkcursor.h:228:12: note: declared here
 GdkCursor* gdk_cursor_new   (GdkCursorType    cursor_type);
            ^
/home/daniel/dev/snappy/snappy-playpen/shotwell/parts/shotwell/build/plugins/common/RESTSupport.vala: In function ‘publishing_rest_support_google_publisher_web_authentication_pane_on_load_started’:
/home/daniel/dev/snappy/snappy-playpen/shotwell/parts/shotwell/build/plugins/common/RESTSupport.vala:779:2: warning: ‘gdk_cursor_new’ is deprecated [-Wdeprecated-declarations]
             pane_widget.get_window().set_cursor(new Gdk.Cursor(Gdk.CursorType.WATCH));
  ^
In file included from /usr/include/gtk-3.0/gdk/gdk.h:34:0,
                 from ../../plugins/shotwell-plugin-dev-1.0.h:15,
                 from RESTSupport.c:15:
/usr/include/gtk-3.0/gdk/gdkcursor.h:228:12: note: declared here
 GdkCursor* gdk_cursor_new   (GdkCursorType    cursor_type);
            ^
cc  -L/home/daniel/dev/snappy/snappy-playpen/shotwell/parts/shotwell/install/lib -L/home/daniel/dev/snappy/snappy-playpen/shotwell/parts/shotwell/install/usr/lib -L/home/daniel/dev/snappy/snappy-playpen/shotwell/parts/shotwell/install/lib/x86_64-linux-gnu -L/home/daniel/dev/snappy/snappy-playpen/shotwell/parts/shotwell/install/usr/lib/x86_64-linux-gnu -shared Resources.o RESTSupport.o -lgee-0.8 -lxml2 -ljson-glib-1.0 -lwebkit2gtk-4.0 -lgtk-3 -lgdk-3 -lpangocairo-1.0 -lpango-1.0 -latk-1.0 -lcairo-gobject -lcairo -lgdk_pixbuf-2.0 -lsoup-2.4 -lgio-2.0 -lgobject-2.0 -ljavascriptcoregtk-4.0 -lglib-2.0  -o libshotwell-plugin-common.so
make[2]: Verzeichnis „/home/daniel/dev/snappy/snappy-playpen/shotwell/parts/shotwell/build/plugins/common“ wird verlassen
make[1]: Verzeichnis „/home/daniel/dev/snappy/snappy-playpen/shotwell/parts/shotwell/build/plugins“ wird verlassen
Makefile:686: die Regel für Ziel „plugins“ scheiterte
make: *** [plugins] Fehler 2
make: *** Auf noch nicht beendete Prozesse wird gewartet …
Command '['/bin/sh', '/tmp/tmp9e4scpll', 'make', '-j4']' returned non-zero exit status 2
daniel@daydream:~/dev/snappy/snappy-playpen/shotwell$ 
```

### PR DESCRIPTION
**Pull request by LyzardKing**
Fix #193 
Added a custom launcher to be able to use the new desktop-gtk3 part.
The custom launcher does the following:
cd in the snap folder (temporary fix)
add pwd to parameters (to open images)
@dholbach
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/ubuntu/snappy-playpen/pull/224%23discussion_r76563057%22%2C%20%22https%3A//github.com/ubuntu/snappy-playpen/pull/224%23issuecomment-243050549%22%2C%20%22https%3A//github.com/ubuntu/snappy-playpen/pull/224%23issuecomment-243058614%22%2C%20%22https%3A//github.com/ubuntu/snappy-playpen/pull/224%23issuecomment-243064890%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ubuntu/snappy-playpen/pull/224%23issuecomment-243050549%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%3A%2B1%3A%20%22%2C%20%22created_at%22%3A%20%222016-08-29T07%3A22%3A18Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1346979%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dholbach%22%7D%7D%2C%20%7B%22body%22%3A%20%22%40dholbach%20%5Cr%5CnWhat%20the%20custom%20launcher%20does%20will%20not%20be%20needed%20in%20the%20future%2C%20when%20shotwell%20completes%20the%20switch%20to%20gresource.%5Cr%5CnIt%27s%20just%20a%20temporary%20fix.%22%2C%20%22created_at%22%3A%20%222016-08-29T08%3A04%3A45Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7113541%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/LyzardKing%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%20%22%2C%20%22created_at%22%3A%20%222016-08-29T08%3A35%3A51Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1346979%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dholbach%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%2001f211a94bb60d5230520434668392ac8d6555c6%20shotwell/shotwell-launcher%207%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ubuntu/snappy-playpen/pull/224%23discussion_r76563057%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Is%20the%20above%20something%20we%20should%20add%20to%20the%20desktop%20launcher%3F%20%40didrocks%20what%20do%20you%20think%3F%22%2C%20%22created_at%22%3A%20%222016-08-29T07%3A21%3A42Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1346979%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dholbach%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20shotwell/shotwell-launcher%3AL1-8%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [ ] <a href='#crh-comment-Pull 01f211a94bb60d5230520434668392ac8d6555c6 shotwell/shotwell-launcher 7'></a> &nbsp;<b><a href='https://github.com/ubuntu/snappy-playpen/pull/224#discussion_r76563057'>File: shotwell/shotwell-launcher:L1-8</a></b>
- <a href='https://github.com/dholbach'></a> Is the above something we should add to the desktop launcher? @didrocks what do you think?
- [x] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/ubuntu/snappy-playpen/pull/224#issuecomment-243050549'>General Comment</a></b>
- <a href='https://github.com/dholbach'></a> :+1:
- <a href='https://github.com/LyzardKing'></a> @dholbach
What the custom launcher does will not be needed in the future, when shotwell completes the switch to gresource.
It's just a temporary fix.

<a href='https://www.codereviewhub.com/ubuntu/snappy-playpen/pull/224?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/ubuntu/snappy-playpen/pull/224?approve=1'></a>&nbsp;<a href='https://github.com/ubuntu/snappy-playpen/pull/224'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by didrocks** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/ubuntu/snappy-playpen/pull/224'>Refresh</a>.<a href='#crh-update-none'></a>

File: shotwell/shotwell-launcher [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#! /bin/bash
+if [ ! -z "$1" ]
+  then
+    pwd=$PWD/
+fi
+cd $SNAP
+desktop-launch shotwell "$pwd$@"
```
Comment (dholbach) [inline shotwell/shotwell-launcher]: Is the above something we should add to the desktop launcher? @didrocks what do you think?

**Comment by dholbach** [comment]
:+1:

**Comment by LyzardKing** [comment]
@dholbach 
What the custom launcher does will not be needed in the future, when shotwell completes the switch to gresource.
It's just a temporary fix.

**Comment by dholbach** [comment]
:+1:

**Pull request merged**